easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,148 @@
+##
+# Copyright 2009-2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing LAMMPS, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+@author: Alexander Schnurpfeil (Juelich Supercomputer Centre)
+"""
+import os
+
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_LAMMPS(EasyBlock):
+    """Support for building/installing LAMMPS."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for LAMMPS."""
+        super(EB_LAMMPS, self).__init__(*args, **kwargs)
+
+        self.build_in_installdir = True
+
+    @staticmethod
+    def extra_options():
+        """Custom easyconfig parameters for LAMMPS."""
+        extra_vars = {
+            'packages': [[], "LAMMPS packages to install", CUSTOM],
+        }
+        return EasyBlock.extra_options(extra_vars)
+
+    def extract_step(self):
+        """Extract LAMMPS sources."""
+        # strip off top-level subdirectory
+        self.cfg.update('unpack_options', '--strip-components=1')
```

Comment:
works fine with upcoming 2.1 and current develop thanks to https://github.com/hpcugent/easybuild-framework/pull/1229

for now, you can change this to what you mention, but that will override any unpack options listed in the easyconfig file; using `update`, you'd append to unpack options specified in the easyconfig instead